### PR TITLE
[TECHNICAL SUPPORT] LPS-161372 Can't select image to create a link in a repeatable Rich text field

### PR DESCRIPTION
**Pull request by NemethNorbert**
Hello,
[LPS-161372](https://issues.liferay.com/browse/LPS-161372)

I followed the suggestion from https://issues.liferay.com/browse/PTR-3322?focusedCommentId=2757239&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2757239
and updated the editorConfig with the new name in case of nested repeated rich_text fields as well.

Please review my changes,

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/reducers/fieldReducer.es.js [hunk 1]
```diff
@@ -222,14 +259,33 @@ export default function fieldReducer(state, action) {
...
 											name,
-											currentField.nestedFields
-										),
-									};
+											nestedFields: updateNestedFieldNames(
+												name,
+												currentField.nestedFields
+											),
+										};
+									}
```
Comment (alinedoleron) [inline modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/reducers/fieldReducer.es.js:288]: Please, apply the same simplification here.

``` javascript
if (currentField.type === 'rich_text') {
  return {
	  ...currentField,
	  ...(currentField.editorConfig && {
		  editorConfig: updateEditorConfigFieldName(
			  currentField.editorConfig,
			  name
		  ),
	  }),
	  name,
	  nestedFields: updateNestedFieldNames(
		  name,
		  currentField.nestedFields
	  ),
  };
}
```

**Review by alinedoleron — CHANGES_REQUESTED** [review]

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/reducers/fieldReducer.es.js [hunk 2]
```diff
@@ -61,18 +61,55 @@ export function updateNestedFieldNames(parentFieldName, nestedFields) {
...
+		if (typeof value === 'string') {
+			const parsedName = parseName(decodeURIComponent(value));
+
+			if (Object.keys(parsedName).length) {
+				const test = encodeURIComponent(generateName(null, parsedName));
+
+				updatedEditorConfig[key] = value.replace(
+					test,
+					encodeURIComponent(name) + 'selectItem'
+				);
```
[the remark this answers is not in the document]
Comment (NemethNorbert) [reply modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/reducers/fieldReducer.es.js]: Hey @alinedoleron,

I renamed the `test` variable to `currentName` and simplified the returns. I looked into the hardcoded `'selectItem'` string and I could not find a better way to do this. On the other hand I also saw many examples in our codebase, where we do the same. I force pushed the change

**Review by alinedoleron — APPROVED** [review]
LGTM!

**Comment by alinedoleron** [comment]
ci:forward

**Comment by alinedoleron** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**